### PR TITLE
Implement user data class and repository interface.

### PR DESCRIPTION
**Pull request by AlexandreBeeDeeBop**
### What is this PR?

This PR introduces a User data class containing all useful informations about a biker on our app. It also introduces a UserRepository interface to manage user-related data operations efficiently for my next PR.

### How?
User data class encapsulates user information such as ID, name, email, favorite parking spots and more.
UserRepository interface defines methods for user data management, enabling future database interactions.

### Why?
To establish a structured foundation for user operations, facilitating easier integration with storage solutions and improving code maintainability.

### Additional details?
A bunch of additional information is planned to be added to the user's data class and a few functions to the interface too, but not yet because of the complications they induce and the lack of time I have for this sprint. They are not essential and it can wait another 1-2 sprints. This PR does not include testing, I just need your feedback on the features I created. They cant break anything on the main. My second PR, to come later, will include testing using 2 mock users and the interface.

### CODE REVIEW

**Comment by AlexandreBeeDeeBop** [comment]
oops i forgot ktfmt format thingy

File: app/src/main/java/com/github/se/cyrcle/model/user/User.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    val lastName: String? = null,
+    val email: String = "",
+    val profilePictureUrl: String? = null,
+    val homeAddress: String? = null,
+    val favoriteParkingSpots: List<String> = emptyList(),
+    // TODO For some reason these break my serialization/
+    // deserialization and i dont have the time to fix it :
+    // val lastKnownLocation: Location? = null,
+    // val lastLoginTime: Timestamp? = null,
+    // val accountCreationDate: Timestamp? = null,
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/model/user/User.kt:16]: Good to have commented these fields for now. We'll have to look into why this happens, but the fields aren't super relevant yet so it's not too big of an issue as of right now

File: app/src/main/java/com/github/se/cyrcle/model/user/User.kt
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/model/user/User.kt:1]: We'll have to look into which we make mandatory for the user to input (for eg maybe a profile picture isn't mandatory?)

File: app/src/main/java/com/github/se/cyrcle/model/user/User.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.github.se.cyrcle.model.user
+
+data class User(
+    val userId: String = "",
```
Comment (sean-prz) [inline app/src/main/java/com/github/se/cyrcle/model/user/User.kt:4]: I do not see why we allow a default value for userId to be the empty string? shouldn't we forbid the creation of User Object if no userId is specified ?

Comment (LouisJamesVasseur) [reply app/src/main/java/com/github/se/cyrcle/model/user/User.kt:4]: Agree with this, but I'm not sure it's too much of an issue at this point in time. As I said, we'll need to figure out what fields are mandatory or not but this is a design decision for a meeting.

File: app/src/main/java/com/github/se/cyrcle/model/user/UserRepository.kt
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/model/user/UserRepository.kt:1]: Everything here seems up to standard for a ViewModel, and well commented

**Review by LouisJamesVasseur — APPROVED** [review]

File: app/src/main/java/com/github/se/cyrcle/model/user/User.kt (hunk 2 above)
Comment (matttsss) [reply app/src/main/java/com/github/se/cyrcle/model/user/User.kt:4]: I kind of agree with Sean, this is kind of crucial for testing

Comment (Pijukai) [reply app/src/main/java/com/github/se/cyrcle/model/user/User.kt]: I strongly agree as well.

**Review by LouisJamesVasseur — CHANGES_REQUESTED** [review]
I think that the userID default value should be fixed ASAP, but the rest looks good

**Review by LouisJamesVasseur — APPROVED** [review]
As the userID issue has been fixed, I'll approve this PR. Thanks for the quick fix!

File: app/src/main/java/com/github/se/cyrcle/model/user/User.kt [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package com.github.se.cyrcle.model.user
+
+data class User(
+    val userId: String,
+    val username: String = "",
+    val firstName: String? = null,
```
Comment (Pijukai) [inline app/src/main/java/com/github/se/cyrcle/model/user/User.kt:6]: I think we should replace `String? = null` by `String = ""` for the other inputs for consistency. We do not expect any other type besides `String` and except for the username they can be empty if the user doesn't want to give corresponding info.

**Review by Pijukai — CHANGES_REQUESTED** [review]
Address all the comments and I approve.

**Review by Pijukai — APPROVED** [review]
All good just pass the CI and you can merge

**Comment by AlexandreBeeDeeBop** [comment]
bruh i already did idk why it didnt work, my bad

**Review by LouisJamesVasseur — APPROVED** [review]
Everything seems good to merge!

**Pull request merged**